armi/physics/fuelCycle/fuelHandlers.py

```diff
@@ -1172,17 +1166,43 @@ def _transferStationaryBlocks(self, assembly1, assembly2):
 
         These blocks in effect are not moved at all.
         """
-        for index in self.cs["stationaryBlocks"]:
-            # this block swap is designed to ensure that all blocks have the
-            # correct parents and children structure at the end of the swaps.
-            tempBlock1 = assembly1[index]
-            assembly1.remove(tempBlock1)
+        # grab stationary block flags
+        sBFList = self.r.core.stationaryBlockFlagsList
+
+        # identify stationary blocks for assembly 1
+        a1StationaryBlocks = [
+            [block, block.spatialLocator.k]
+            for block in assembly1
+            if any(block.hasFlags(sbf) for sbf in sBFList)
+        ]
+        # identify stationary blocks for assembly 2
+        a2StationaryBlocks = [
+            [block, block.spatialLocator.k]
+            for block in assembly2
+            if any(block.hasFlags(sbf) for sbf in sBFList)
+        ]
 
-            tempBlock2 = assembly2[index]
-            assembly2.remove(tempBlock2)
+        # check for any inconsistencies in stationary blocks and ensure alignment
+        if [block[1] for block in a1StationaryBlocks] != [
+            block[1] for block in a2StationaryBlocks
+        ]:
+            raise ValueError(
+                """Different number and/or locations of stationary blocks 
+                 between {} and {}.""".format(
+                    assembly1, assembly2
```

Comment:
It would be a better error to present the user with which specific stationary blocks are mismatched by also including the the print out of `a1StationaryBlocks` and `a2StationaryBlocks` so that the user isn't left wondering what to do with this failure.

Reply:
Now prints out the stationary blocks